desktop/core/src/desktop/lib/botserver/views_tests.py

```diff
@@ -75,49 +77,58 @@ def test_handle_on_link_shared(self):
-            assert_raises(PopupException, handle_on_link_shared, "channel", "12.1", [{"url": inv_gist_url}])
-
-            # Document does not exist
-            document2_objects_get.side_effect = PopupException('message')
-            _get_gist_document.side_effect = PopupException('message')
-            
-            qhistory_url = "https://demo.gethue.com/hue/editor?editor=109644"
-            gist_url = "https://demo.gethue.com/hue/gist?uuid=6d1c407b-d999-4dfd-ad23-d3a46c19a427"
-            assert_raises(PopupException, handle_on_link_shared, "channel", "12.1", [{"url": qhistory_url}])
-            assert_raises(PopupException, handle_on_link_shared, "channel", "12.1", [{"url": gist_url}])
\ No newline at end of file
+            with patch('desktop.lib.botserver.views.send_result_file') as send_result_file:
+
+              client = make_logged_in_client(username="test", groupname="default", recreate=True, is_superuser=False)
+              user = User.objects.get(username="test")
+              channel_id = "channel"
+              message_ts = "12.1"
+
+              # qhistory link
+              links = [{"url": "https://demo.gethue.com/hue/editor?editor=123456"}]
+              doc_data = {
+                "dialect": "mysql",
+                "snippets": [{
+                  "database": "hue",
+                  "statement_raw": "SELECT 5000",
+                }]
+              }
+              document2_objects_get.return_value = Mock(data=json.dumps(doc_data), owner=user)
+              mock_unfurl_payload.return_value = {
+                'payload': {},
+                'file_status': True,
+              }
+              handle_on_link_shared(channel_id, message_ts, links)
+              assert_true(chat_unfurl.called)
+              assert_true(send_result_file.called)
+
+              # gist link
+              doc_data = {"statement_raw": "SELECT 98765"}
+              _get_gist_document.return_value = Mock(data=json.dumps(doc_data), owner=user, extra='mysql')
+              links = [{"url": "http://demo.gethue.com/hue/gist?uuid=random"}]
+
+              mock_unfurl_payload.return_value = {
+                'payload': {},
+                'file_status': False,
+              }
+              handle_on_link_shared(channel_id, message_ts, links)
+              assert_true(chat_unfurl.called)
+
+              # Cannot unfurl link with invalid links
+              inv_qhistory_url = "https://demo.gethue.com/hue/editor/?type=4"
+              inv_gist_url = "http://demo.gethue.com/hue/gist?uuids/=xyz"
+              assert_raises(PopupException, handle_on_link_shared, "channel", "12.1", [{"url": inv_qhistory_url}])
+              assert_raises(PopupException, handle_on_link_shared, "channel", "12.1", [{"url": inv_gist_url}])
+
+              # Document does not exist
+              document2_objects_get.side_effect = PopupException('message')
```

Comment:
Here and below can't raise DoesNotExist exception instead? And something better than 'message'?